src/pages/TokenDetails/index.tsx

```diff
@@ -79,10 +80,29 @@ export default function TokenDetails() {
   const location = useLocation()
   const { tokenAddress } = useParams<{ tokenAddress?: string }>()
   const token = useToken(tokenAddress)
+  const tokenWarning = tokenAddress ? checkWarning(tokenAddress) : null
+  const isBlockedToken = !!tokenWarning && !tokenWarning?.canProceed
+  const navigate = useNavigate()
+
+  const [continueSwap, setContinueSwap] = useState<{ resolve: (value: boolean | PromiseLike<boolean>) => void }>()
+  const shouldShowSpeedbump = !useIsUserAddedToken(token) && !!tokenWarning
+  // Show token safety modal if Swap-reviewing a warning token, at all times if the current token is blocked
+
```

Comment:
rm newline so the comment is clearly for `onSpeedbump` (or move the comment to the fn defn above L92)